monai/utils/pixel_nms.py

```diff
@@ -0,0 +1,83 @@
+from typing import Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.networks.layers import GaussianFilter
+
+
+class PixelNMS:
+    """
+    Performs pixel based non-maximum suppression (NMS) on the probabilities map via
+    iteratively selecting the coordinate with highest probability and then move it as well
+    as its surrounding values. The remove range is determined by the parameter `pixel_dis`.
+    If multiple coordinates have the same highest probability, only one of them will be
+    selected.
+
+    Args:
+        spatial_dims: number of spatial dimensions of the input probabilities map.
+        sigma: the standard deviation for gaussian filter.
+            It could be a single value, or `spatial_dims` number of values. Defaults to 0.
+        prob_threshold: the probability threshold, the PixelNMS will stop searching if
+            the highest probability is no larger than the threshold. Defaults to 0.5.
+        pixel_dis: the distance in pixels that determines the surrounding area of the selected
+            coordinate that will be removed before the next iteration. The area is a square (cube)
+            for 2D (3D) input, and the edge length is 2 * pixel_dis. Defaults to 24.
+
+    Return:
+        a list of selected lists, where inner lists contain probability and coordinates.
+        For example, for 3D input, the inner lists are in the form of [probability, x, y, z].
+
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0,
+        prob_threshold: float = 0.5,
+        pixel_dis: int = 24,
```

Comment:
```suggestion
        box_size: int = 48,
```
To make this NMS more generic, could you please change ``pixel_dis` to `box_size` and use a multi-dimension (the same number of dimension as `spatial_dims`). So it would allow for ` box_size = (10, 20)`, ` box_size = (10, 20, 30)`, and if one number is provided makes it tuple of length `spatial_dims`,

Reply:
Just be aware that `box_size` at each dimension is twice as `radius`.